src/base/log.cpp

```
@@ -330,32 +330,20 @@ class CWindowsConsoleLogger : public ILogger
 	}
 	void Log(const CLogMessage *pMessage) override
 	{
-		wchar_t *pWide = (wchar_t *)malloc((pMessage->m_LineLength + 2) * sizeof(*pWide));
-		const char *p = pMessage->m_aLine;
-		int WLen = 0;
+		int WLen = MultiByteToWideChar(CP_UTF8, 0, pMessage->m_aLine, pMessage->m_LineLength, NULL, 0);
+		if(!WLen)
+			return;
 
-		mem_zero(pWide, (pMessage->m_LineLength + 2) * sizeof(*pWide));
-
-		for(int Codepoint = 0; (Codepoint = str_utf8_decode(&p)); WLen++)
+		WCHAR *pWide = (WCHAR *)malloc((WLen + 2) * sizeof(*pWide));
+		WLen = MultiByteToWideChar(CP_UTF8, 0, pMessage->m_aLine, pMessage->m_LineLength, pWide, WLen);
+		if(!WLen)
```

I don't think the error should be ignored. I think either crashing or printing something would be better.

---

Now it falls back to printing the log message in ANSI mode (with broken unicode) when `MultiByteToWideChar` fails. Seems more useful to me than printing a relevant error message (which could also fail) or crashing.

---

Crashing is useful if the error shouldn't actually happen, because it'd allow us to notice it. Same for printing an error message instead (which can't fail the conversion because we can hardcode it in UTF-16).

Silently falling back to a broken conversion is bad though, we won't notice the bug and it might hide other bugs and make stuff harder to debug.